src/rgw/rgw_acl.cc

```diff
@@ -52,7 +52,20 @@ void RGWAccessControlList::add_grant(ACLGrant *grant)
   _add_grant(grant);
 }
 
-uint32_t RGWAccessControlList::get_perm(const DoutPrefixProvider* dpp, 
+void RGWAccessControlList::remove_canon_user_grant(rgw_user& user_id)
+{
+  auto multi_map_iter = grant_map.find(user_id.to_str());
+  if(multi_map_iter != grant_map.end()) {
```

Comment:
the user could have several grants in this multimap. since we're erasing their permissions from `acl_user_map`, i guess we should remove all matching grants too?
```c++
auto grants = grant_map.equal_range(user_id.to_str());
grant_map.erase(grants.first, grants.second);
```
(if there are no matches, `grants` will be an empty range and `erase()` will be a noop)